### PR TITLE
add example dag for dataset_alias

### PR DESCRIPTION
**Pull request by Lee-W**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/example_dags/example_dataset_alias.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from __future__ import annotations
+
+import pendulum
+
+from airflow import DAG
+from airflow.datasets import Dataset, DatasetAlias
+from airflow.decorators import task
+
+with DAG(
+    dag_id="dataset-producer",
```
Comment (jscheffl) [inline airflow/example_dags/example_dataset_alias.py]: All other example DAGs use underscore, can you make it consistent to these?
```suggestion
    dag_id="dataset_producer",
```

By the way, there is already one example DAG containing "dataset_produces_1" as DAG name - adding another producer might be confusing. Could you merge it with the other example or name it different in a way that it is easy to distinguish?

File: airflow/example_dags/example_dataset_alias.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+):
+
+    @task(outlets=[Dataset("s3://bucket/my-task")])
+    def produce_dataset_events():
+        pass
+
+    produce_dataset_events()
+
+with DAG(
+    dag_id="dataset-alias-producer",
```
Comment (jscheffl) [inline airflow/example_dags/example_dataset_alias.py]: Same here
```suggestion
    dag_id="dataset_alias_producer",
```

File: airflow/example_dags/example_dataset_alias.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    @task(outlets=[DatasetAlias("example-alias")])
+    def produce_dataset_events_through_dataset_alias(*, outlet_events=None):
+        bucket_name = "bucket"
+        object_path = "my-task"
+        outlet_events["example-alias"].add(Dataset(f"s3://{bucket_name}/{object_path}"))
+
+    produce_dataset_events_through_dataset_alias()
+
+with DAG(
+    dag_id="dataset-consumer",
```
Comment (jscheffl) [inline airflow/example_dags/example_dataset_alias.py]: +1
```suggestion
    dag_id="dataset_consumer",
```

File: airflow/example_dags/example_dataset_alias.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+):
+
+    @task
+    def consume_dataset_event():
+        pass
+
+    consume_dataset_event()
+
+with DAG(
+    dag_id="dataset-alias-consumer",
```
Comment (jscheffl) [inline airflow/example_dags/example_dataset_alias.py]: +1
```suggestion
    dag_id="dataset_alias_consumer",
```

File: airflow/example_dags/example_dataset_alias.py (hunk 1 above)
Comment (Lee-W) [reply airflow/example_dags/example_dataset_alias.py]: I think naming it differently might be better. I'll fix it. Thanks for your suggestion!

Comment (Lee-W) [reply airflow/example_dags/example_dataset_alias.py]: Just updated it

File: airflow/example_dags/example_dataset_alias.py (hunk 2 above)
Comment (Lee-W) [reply airflow/example_dags/example_dataset_alias.py]: updated it with an underscore. Thanks!

File: airflow/example_dags/example_dataset_alias.py (hunk 3 above)
Comment (Lee-W) [reply airflow/example_dags/example_dataset_alias.py]: updated it with an underscore. Thanks!

File: airflow/example_dags/example_dataset_alias.py (hunk 4 above)
Comment (Lee-W) [reply airflow/example_dags/example_dataset_alias.py]: updated it with an underscore. Thanks!

**Review by phanikumv — APPROVED** [review]

**Pull request merged**